skunkworks_crow/src/main/java/org/odk/share/activities/ScannerActivity.java

```diff
@@ -0,0 +1,114 @@
+package org.odk.share.activities;
+
+/*
+ * created by Chromicle
+ */
+
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.view.KeyEvent;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.Button;
+
+import com.journeyapps.barcodescanner.CaptureManager;
+import com.journeyapps.barcodescanner.DecoratedBarcodeView;
+
+import org.odk.share.R;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+
+public class ScannerActivity extends AppCompatActivity implements
+        DecoratedBarcodeView.TorchListener {
+
+    private CaptureManager capture;
+    private DecoratedBarcodeView decoratedScannerView;
+    private boolean isFlashLightOn;
+
+    @BindView(R.id.switch_flashlight)
+    Button toggleFlashlightButton;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_scanner);
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
+
+        decoratedScannerView = findViewById(R.id.zxing_barcode_scanner);
+        decoratedScannerView.setTorchListener(this);
+        ButterKnife.bind(this);
+
+        capture = new CaptureManager(this, decoratedScannerView);
+        capture.initializeFromIntent(getIntent(), savedInstanceState);
+        capture.decode();
+    }
+
+    private boolean hasFlash() {
+        return getApplicationContext().getPackageManager()
+                .hasSystemFeature(PackageManager.FEATURE_CAMERA_FLASH);
+    }
+
+    public void switchFlashlight() {
+        if (isFlashLightOn) {
+            decoratedScannerView.setTorchOff();
+            isFlashLightOn = false;
+        } else {
+            decoratedScannerView.setTorchOn();
+            isFlashLightOn = true;
+        }
+    }
+
+    @Override
+    public void onTorchOn() {
+        toggleFlashlightButton.setBackgroundResource(R.drawable.ic_flash_white_on);
+    }
+
+    @Override
+    public void onTorchOff() {
+        toggleFlashlightButton.setBackgroundResource(R.drawable.ic_flash_white_off);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        capture.onResume();
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        capture.onPause();
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        capture.onDestroy();
+    }
+
+    @Override
+    protected void onSaveInstanceState(@NonNull Bundle outState) {
+        super.onSaveInstanceState(outState);
+        capture.onSaveInstanceState(outState);
+    }
+
+    @OnClick(R.id.switch_flashlight)
+    public void toggleButton() {
+        if (!hasFlash()) {
+            toggleFlashlightButton.setVisibility(View.GONE);
```

Comment:
The check for hiding the button should happen in `onCreate` method

Reply:
Yeah I updated